gnovm/pkg/gnolang/preprocess.go

```diff
@@ -2449,16 +2449,33 @@ func evalStaticTypeOf(store Store, last BlockNode, x Expr) Type {
 	t := evalStaticTypeOfRaw(store, last, x)
 	if tt, ok := t.(*tupleType); ok {
 		if len(tt.Elts) != 1 {
-			panic(fmt.Sprintf(
-				"evalStaticTypeOf() only supports *CallExpr with 1 result, got %s",
-				tt.String(),
-			))
-		} else {
-			return tt.Elts[0]
+			loc := last.GetLocation()
+			funcName := getFunctionName(x)
+			valueCount := len(tt.Elts)
+
+			var msg string
+			if valueCount == 0 {
+				msg = fmt.Sprintf("%s: %s() (no value) used as value", loc, funcName)
+			} else {
+				msg = fmt.Sprintf("%s: %s() (%d values) used as single value", loc, funcName, valueCount)
+			}
+
+			panic(fmt.Errorf("%s\nHint: Ensure the function returns a single value, or use multiple assignment", msg))
```

Comment:
This hint may not be very effective if it corresponds to one of the messages above, such as:

panic: main/func.gno:3:1: f() (no value) used as value
Hint: Ensure the function returns a single value, or use multiple assignment.

I mean in this case, "or use multiple assignment" is superfluous。

Reply:
Or not, this check should be checked ahead of `evalStaticTypeOf` somewhere?

Reply:
well I think this part feels odd to me(the original logic in `evalStaticTypeOf`:

```go
if len(tt.Elts) != 1 {
			panic(fmt.Sprintf(
				"evalStaticTypeOf() only supports *CallExpr with 1 result, got %s",
				tt.String(),
			))
```

at the very least for case like:
```go
n := f2()

func f2() (string, int) {
	return "hey", 0
}
```
the check can happened in *AssignStmt, to identify a mismatch on LHS and RHS.

Reply:
Writing so much just for the sake of discussion.

Reply:
Yes @ltzmaxwell I think this check should be done in AssignStmt and ValueDecl in trans_enter